### PR TITLE
[fix] twitter to salesforce fix

### PR DESCRIPTION
**Pull request by mkralik3**
//test: @integrations-twitter-to-salesforce

### CODE REVIEW

**Comment by mkralik3** [comment]
Blocked by https://github.com/syndesisio/syndesis-qe/pull/620/commits/876542d36ac83b069686ce6de39f1e1ebabde771

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by tplevko** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: ui-tests/src/test/resources/features/integrations/twitter-to-salesforce.feature [hunk 1]
```diff
@@ -74,9 +75,10 @@ Feature: Integration - Twitter to Salesforce
     Then wait until integration "Twitter to Salesforce E2E" gets into "Running" state
     #And verify s2i build of integration "Twitter to Salesforce E2E" was finished in duration 1 min
 
-    Then tweet a message from twitter_talky to "Twitter Listener" with text "Red Hat"
+    Then tweet a message from twitter_talky to "Twitter Listener" with text "Red Hat #e2e #syndesis4ever"
+    And sleep for "30000" ms
 
     # validate salesforce contacts
-    Then check that contact from SF with last name: "Jackson" has description "Red Hat"
+    Then check that contact from SF with last name: "Jackson" has description "Red Hat #e2e #syndesis4ever"
```
Comment (tplevko) [inline ui-tests/src/test/resources/features/integrations/twitter-to-salesforce.feature]: Seems like there is an issue in this validation step. The last name of the contact created in the salesforce is not really "Jackson" but "Talky". The second issue is, that the description will also contain the "@syndesis_listen" at the end.

File: ui-tests/src/test/resources/features/integrations/twitter-to-salesforce.feature [hunk 2]
```diff
@@ -74,9 +75,10 @@ Feature: Integration - Twitter to Salesforce
...
     # clean-up in salesforce
     Then delete contact from SF with last name: "Jackson"
```
Comment (tplevko) [inline ui-tests/src/test/resources/features/integrations/twitter-to-salesforce.feature]: Also this validation step is not OK, for reason mentioned above.

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mkralik3** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mkralik3** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: ui-tests/src/test/resources/features/integrations/twitter-to-salesforce.feature [hunk 3]
```diff
@@ -74,9 +75,17 @@ Feature: Integration - Twitter to Salesforce
     Then wait until integration "Twitter to Salesforce E2E" gets into "Running" state
     #And verify s2i build of integration "Twitter to Salesforce E2E" was finished in duration 1 min
 
-    Then tweet a message from twitter_talky to "Twitter Listener" with text "Red Hat"
+    Then tweet a message from twitter_talky to "Twitter Listener" with text "Red Hat #syndesis4ever"
```
Comment (jstastny-cz) [inline ui-tests/src/test/resources/features/integrations/twitter-to-salesforce.feature]: Then - And - Then
->
Then - And - And

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java [hunk 4]
```diff
@@ -174,12 +174,18 @@ public void validateIntegration(String twAccount, String record) {
...
-    @Then("check SF does not contain contact for tw accound: \"([^\"]*)\"")
+    @Then("check SF does not contain contact for tw account: \"([^\"]*)\"")
     public void checkSfDoesNotContain(String twAccount) {
         assertThat(getSalesforceContact(salesforce, accountsDirectory.getAccount(twAccount)
                 .get().getProperty("screenName")).isPresent()).isEqualTo(false);
     }
 
+    @Then("check SF contains contact for tw account: \"([^\"]*)\"")
+    public void checkSfDoesContain(String twAccount) {
+        assertThat(getSalesforceContact(salesforce, accountsDirectory.getAccount(twAccount)
```
Comment (jstastny-cz) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java]: Wouldn't be a private method with boolean argument be better than code replication?

Comment (mkralik3) [reply utilities/src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java]: yes, it would be better

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by tplevko** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by tplevko** [comment]
@mkralik3 can you please add this to the 1.5.x. branch too?

**Pull request merged**